core/charm/repository/sanitizer.go

```diff
@@ -40,5 +40,12 @@ func sanitizeCharmOrigin(received, requested corecharm.Origin) (corecharm.Origin
 	}
 	result.Platform.OS = strings.ToLower(result.Platform.OS)
 
+	// Another problem area is the origin channel. We desire a full channel (track and risk)
+	// but if the effective channel implicitly resolves to a 'latest' channel, charmhub will
+	// just return the risk. This only happens for 'latest'. If we have no track, it must be latest
+	if result.Channel != nil && result.Channel.Track == "" {
+		result.Channel.Track = "latest"
```

Comment:
Should this not be `result.Channel.Risk = "stable"`?

Reply:
At this point, if we leave the track empty and the charm author adds a default track you could end up in an unexpected place.